### PR TITLE
[Feature] Add histogram statistics Cache Loader

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

**Review by Seaven — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 80 / 198 (40.40%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/Bucket.java|0|10|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CsvFileStatisticsStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/Histogram.java|0|4|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/MockTpchStatisticStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/EmptyStatisticStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnHistogramStatsCacheLoader.java|3|48|06.25%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java|18|43|41.86%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnBasicStatsCacheLoader.java|49|77|63.64%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnStatsCacheKey.java|10|13|76.92%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 80 / 198 (40.40%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/Bucket.java|0|10|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CsvFileStatisticsStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/Histogram.java|0|4|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/MockTpchStatisticStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/EmptyStatisticStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnHistogramStatsCacheLoader.java|3|48|06.25%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java|18|43|41.86%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnBasicStatsCacheLoader.java|49|77|63.64%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnStatsCacheKey.java|10|13|76.92%|

**Comment by wanpengfei-git** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 80 / 198 (40.40%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/Bucket.java|0|10|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CsvFileStatisticsStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/Histogram.java|0|4|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/MockTpchStatisticStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/EmptyStatisticStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnHistogramStatsCacheLoader.java|3|48|06.25%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java|18|43|41.86%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnBasicStatsCacheLoader.java|49|77|63.64%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnStatsCacheKey.java|10|13|76.92%|

**Review by Seaven — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/ColumnStatsCacheKey.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        ColumnStatsCacheKey cacheKey = (ColumnStatsCacheKey) o;
+
+        if (tableId != cacheKey.tableId) {
+            return false;
+        }
+        return column.equals(cacheKey.column);
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/ColumnStatsCacheKey.java:29]: equalsIgnoreCase

**Review by kangkaisen — APPROVED** [review]

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 80 / 198 (40.40%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/Bucket.java|0|10|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CsvFileStatisticsStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/Histogram.java|0|4|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/MockTpchStatisticStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/EmptyStatisticStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnHistogramStatsCacheLoader.java|3|48|06.25%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java|18|43|41.86%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnBasicStatsCacheLoader.java|49|77|63.64%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnStatsCacheKey.java|10|13|76.92%|

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 80 / 198 (40.40%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/Bucket.java|0|10|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CsvFileStatisticsStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/Histogram.java|0|4|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/MockTpchStatisticStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/EmptyStatisticStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnHistogramStatsCacheLoader.java|3|48|06.25%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java|18|43|41.86%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnBasicStatsCacheLoader.java|49|77|63.64%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnStatsCacheKey.java|10|13|76.92%|

**Review by Seaven — APPROVED** [review]

**Pull request merged**